File: frontend/src/api/project.ts
import { Project, Session } from '../types';
import { request } from './client';

export async function fetchProjects(userId: string): Promise<Project[]> {
  const data = await request<Project[]>(`/api/projects?user_id=${userId}`);
  return data;
}

export async function createProject(
  userId: string,
  name: string,
  description = '',
  metadata: Record<string, string> = {}
): Promise<{ id: string; status: string }> {
  const data = await request<{ id: string; status: string }>('/api/projects', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ user_id: userId, name, description, metadata }),
  });
  return data;
}

export async function updateProject(
  projectId: string,
  name: string,
  description = '',
  metadata: Record<string, string> = {}
): Promise<void> {
  await request<void>(`/api/projects/${projectId}`, {
    method: 'PUT',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ name, description, metadata }),
  });
}

export async function deleteProject(projectId: string): Promise<void> {
  await request<void>(`/api/projects/${projectId}`, {
    method: 'DELETE',
  });
}

export async function fetchProjectSessions(projectId: string): Promise<Session[]> {
  const data = await request<Session[]>(`/api/projects/${projectId}/sessions`);
  return data;
}

File: cmd/http/dto/project.go
package dto

type CreatProjectReq struct {
	UserID      string            `json:"user_id"`
	Name        string            `json:"name"`
	Description string            `json:"description"`
	Metadata    map[string]string `json:"metadata"`
}

type ProjectUserIdReq struct {
	UserID string `json:"user_id" form:"user_id"`
}

type ProjectIdReq struct {
	ProjectId string `uri:"id"`
}

type ProjectUpdateReq struct {
	ProjectId   string `uri:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
}

type ProjectResp struct {
	ID     string `json:"id"`
	Status string `json:"status"`
}
